lib/resourcemerge/clusterapi.go

```diff
@@ -14,5 +17,70 @@ func EnsureMachineSet(modified *bool, existing *clusterv1.MachineSet, required *
 // ensureMachineSetSpec ensures that the existing matches the required.
 // modified is set to true when existing had to be updated with required.
 func ensureMachineSetSpec(modified *bool, existing *clusterv1.MachineSetSpec, required clusterv1.MachineSetSpec) {
-	// TODO(alberto): implement this
+	setInt32IfSet(modified, existing.Replicas, *required.Replicas)
+	if !equality.Semantic.DeepEqual(existing.Selector, required.Selector) {
+		*modified = true
+		existing.Selector = required.Selector
+	}
+	EnsureObjectMeta(modified, &existing.Template.Spec.ObjectMeta, required.Template.Spec.ObjectMeta)
+	if !equality.Semantic.DeepEqual(existing.Template.Spec.Taints, required.Template.Spec.Taints) {
+		*modified = true
+		existing.Template.Spec.Taints = required.Template.Spec.Taints
+	}
+	// TODO(vikasc): verify if other fields also needs to be synced
+}
+
+// EnsureCluster ensures that the existing matches the required.
+// modified is set to true when existing had to be updated with required.
+func EnsureCluster(modified *bool, existing *clusterv1.Cluster, required *clusterv1.Cluster) {
+	EnsureObjectMeta(modified, &existing.ObjectMeta, required.ObjectMeta)
+	ensureClusterSpec(modified, &existing.Spec, required.Spec)
+}
+
+// ensureClusterSpec ensures that the existing matches the required.
+// modified is set to true when existing had to be updated with required.
+func ensureClusterSpec(modified *bool, existing *clusterv1.ClusterSpec, required clusterv1.ClusterSpec) {
+	for _, required := range required.ClusterNetwork.Services.CIDRBlocks {
```

Comment:
to clarify: we are only leveraging the machineAPI, the object cluster is needed at the moment because they are loosely coupled on the machine controller and the actuator interface